### PR TITLE
[Core] add more openmp utils

### PR DESCRIPTION
**Pull request by ddiezrod**
**Description**
Adding wrappers for more openmp utils. This was added by @jrubiogonzalez some time ago, but we never merged it back here. As it is my turn to merge master in @KratosMultiphysics/altair code, I want to take the chance to solve this conflict-

**Changelog**
-Added more openmp functions in open mp utils.

**Additional info**

### CODE REVIEW

**Comment by philbucher** [comment]
I will not block bcs I realize updating this on your side is painful, but this kinda goes against the using the parallel utilities

**Comment by ddiezrod** [comment]
The aim of this PR is mainly to fix the conflict with our side... If there is a better way to do this I can try it. In any case, I would say the only method we really need from these is the GetNumberOfProcessors as it is printed in out .out file.

**Comment by philbucher** [comment]
I don't know the details of your code, but maybe those functions could be added to the parallel utils right away. Then it wouldn't be necessary to update them in the future

**Comment by RiccardoRossi** [comment]
+1 for Philipp's comment...

other than that i am ok

**Review by RiccardoRossi — DISMISSED** [review]

**Review by roigcarlo — CHANGES_REQUESTED** [review]
I have several questions before approving:

1 - Why do you need `omp_get_max_threads` and specially, why do you need to export that to python. That should only be called from inside a parallel region in case you want to know the number of threads if they are not the max (which is a bad practice imo). Also notice that the way this is exported will always return 1 in python, as you are not calling the function from inside a parallel region, nor the utility has a parallel region in the implementation.

2 - Similar question with `omp_get_num_procs`. That will return the maximum number of threads available to the device. Why do you need that? Again we should only use the number of threads assigned to the process (wether they are more, less or exactly equal to the number of threads available in the machine), not the ones available by default. In the best case, you might be requesting resources that are allocated by other processes...

3 - `IsInParallel` also have no use to be exported directly to python. The result will be false, as it is not in parallel region.

We can discuss about 2, but 1 and 3 should be out of the python interface or rewrote so the wrapped call uses a method that queries the value that you want inside a parallel region.

File: kratos/utilities/openmp_utils.h [hunk 1]
```diff
@@ -69,6 +69,44 @@ class OpenMPUtils
...
+    /// Wrapper for omp_get_dynamic().
+    /**
+     @return Dynamic teams are enabled.
+     */
+    static int IsDynamic()
+    {
+#ifdef _OPENMP
+        return omp_get_dynamic();
+#else
+        return 1;
```
Comment (roigcarlo) [inline kratos/utilities/openmp_utils.h]: This should be 0 if in serial.

File: kratos/python/add_other_utilities_to_python.cpp [hunk 2]
```diff
@@ -466,11 +466,12 @@ void AddOtherUtilitiesToPython(pybind11::module &m)
...
         .def(py::init<>())
         .def_static("SetNumThreads", &OpenMPUtils::SetNumThreads)
-    //     .staticmethod("SetNumThreads")
         .def_static("GetNumThreads", &OpenMPUtils::GetNumThreads)
-    //     .staticmethod("GetNumThreads")
         .def_static("PrintOMPInfo", &OpenMPUtils::PrintOMPInfo)
-    //     .staticmethod("PrintOMPInfo")
+        .def_static("GetCurrentNumberOfThreads", &OpenMPUtils::GetCurrentNumberOfThreads)
+        .def_static("GetNumberOfProcessors", &OpenMPUtils::GetNumberOfProcessors)
+        .def_static("IsDynamic", &OpenMPUtils::IsDynamic)
```
Comment (roigcarlo) [inline kratos/python/add_other_utilities_to_python.cpp]: Also no sense to export this directly from the utils, it will return the default value (false)

**Comment by roigcarlo** [comment]
Ok just read about the print of the processes, forget about point 2

File: kratos/python/add_other_utilities_to_python.cpp (hunk 2 above)
Comment (ddiezrod) [reply kratos/python/add_other_utilities_to_python.cpp]: done

File: kratos/utilities/openmp_utils.h (hunk 1 above)
Comment (ddiezrod) [reply kratos/utilities/openmp_utils.h]: done

**Comment by ddiezrod** [comment]
@roigcarlo concerns addressed, another example of why we should try to keep our code as close to master as possible. :) Thanks! Is this ok now?

**Review by roigcarlo — APPROVED** [review]

**Comment by roigcarlo** [comment]
All good!

**Comment by ddiezrod** [comment]
Thanks, I'll try to get rid of this if another option is available in the future.

**Pull request merged**